src/main/kotlin/com/example/passmanager/web/controller/PassController.kt

```diff
@@ -24,32 +25,32 @@ internal class PassController(
 ) {
 
     @GetMapping("/{id}")
-    fun findById(@PathVariable id: String): ResponseEntity<PassDto> {
-        val passById = passService.findById(id)
-        return passById?.run { ResponseEntity.ok(toDto()) } ?: ResponseEntity.notFound().build()
+    fun findById(@PathVariable id: String): Mono<ResponseEntity<PassDto>> {
+        return passService.findById(id)
+            .map { it.toDto() }
+            .map { ResponseEntity.ok(it) }
+            .defaultIfEmpty(ResponseEntity.notFound().build())
     }
 
     @PostMapping
-    fun create(@Valid @RequestBody passDto: PassDto): ResponseEntity<PassDto> {
-        val created = passService.create(passDto.toEntity(), passDto.passOwnerId, passDto.passTypeId)
-        return ResponseEntity.status(HttpStatus.CREATED).body(created.toDto())
+    fun create(@Valid @RequestBody passDto: PassDto): Mono<ResponseEntity<PassDto>> {
+        return passService.create(passDto.toEntity(), passDto.passOwnerId, passDto.passTypeId)
+            .map { it.toDto() }
+            .map { ResponseEntity.status(HttpStatus.CREATED).body(it) }
```

Comment:
can we use annotation for the status code instead of constructing the `ResponseEntity`?

So, the code will be simpler